### ISSUE TITLE
Call If func twice case `ExpressionAttributeNames unused`  ValidationException

### ISSUE THREAD

**Issue by Edwardsj** [guregu/dynamo#89]
Hi, @guregu 
Call If function two or more times in write action like update will case ValidationException like `ValidationException: Value provided in ExpressionAttributeNames unused in expressions: keys: {#sKF2WC3TUNF2HS}`
example:
 ```
	err = table.Update("PlayerID",PlayerID).
		Range("TypeID",TypeID).Add("Quantity",1).
		If("'Quantity' > ?",2).
		If("'TestSt' > ?","aaaa").
		Run()
	
	if err != nil {
		fmt.Println(err)
		return
	}
```
here is the input info
```
{
  ConditionExpression: "#sKRSXG5CTOQ > :v2",
  ExpressionAttributeNames: {
    #sKF2WC3TUNF2HS: "Quantity",
    #sKRSXG5CTOQ: "TestSt"
  },
  ExpressionAttributeValues: {
    :v0: {
      N: "1"
    },
    :v1: {
      N: "2"
    },
    :v2: {
      S: "aaaa"
    }
  },
  Key: {
    PlayerID: {
      S: "1"
    },
    TypeID: {
      N: "2"
    }
  },
  ReturnValues: "NONE",
  TableName: "Test",
  UpdateExpression: "ADD Quantity :v0"
}
```

You can see the problem is the secord condition covered the first one.
update.go line 211
```
func (u *Update) If(expr string, args ...interface{}) *Update {
	cond, err := u.subExpr(expr, args...)
	u.setError(err)
	u.condition = cond    // here !
	return u
}
```
I try to fix it like this, but it's a bit ugly. Could you help to fix it?  ;)
```
func (u *Update) If(expr string, args ...interface{}) *Update {
	cond, err := u.subExpr(expr, args...)
	u.setError(err)
	if u.condition != "" {
		u.condition += " and " + "(" + cond + ")"
	} else {
		u.condition = "(" + cond + ")"
	}
	return u
}
```

**Comment by guregu** [comment]
Hello, thanks a lot for the report. I think your solution is totally fine but I'd like to add some comic logic to this and the other functions that take `If` so I'll break it up a little bit. I'll add these changes soon!

**Comment by guregu** [comment]
I've implemented this and released v1.2.0! Let me know if you have any other issues.
